src/java.base/share/classes/java/lang/invoke/MethodType.java

```diff
@@ -1223,15 +1225,15 @@ static MethodType fromDescriptor(String descriptor, ClassLoader loader)
      * Returns a descriptor string for the method type.  This method
      * is equivalent to calling {@link #descriptorString() MethodType::descriptorString}.
      *
-     * <p>
-     * Note that this is not a strict inverse of {@link #fromMethodDescriptorString fromMethodDescriptorString}.
-     * Two distinct classes which share a common name but have different class loaders
-     * will appear identical when viewed within descriptor strings.
+     * @apiNote
+     * This is not a strict inverse of {@link #fromMethodDescriptorString
+     * fromMethodDescriptorString} which requires a method type descriptor
+     * (JVMS {@jvms 4.3.3}) and a suitable class loader argument. Two distinct
+     * classes which share a common name but have different class loaders will
+     * appear identical when viewed within descriptor strings.
```

Comment:
"will appear identical when viewed within descriptor strings".  Would it be clearer to say that have equal descriptor strings?

Reply:
How about this:
"A method type produced by changing a component type to a distinct class with the same name but a different class loader will have the same descriptor string"